test/playlist-loader.test.js

```diff
@@ -1478,6 +1478,78 @@ QUnit.test('clears the update timeout when switching quality', function(assert)
   assert.equal(1, refreshes, 'only one refresh was triggered');
 });
 
+QUnit.test('Increments retries on playlist error', function(assert) {
+  const loader = new PlaylistLoader('manifest/master.m3u8', this.fakeVhs);
+
+  loader.load();
+
+  // master
+  this.requests.shift().respond(
+    200, null,
+    '#EXTM3U\n' +
+                                '#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=17\n' +
+                                'playlist/playlist.m3u8\n' +
+                                '#EXT-X-STREAM-INF:PROGRAM-ID=2,BANDWIDTH=170\n' +
+                                'playlist/playlist2.m3u8\n' +
+                                '#EXT-X-ENDLIST\n'
+  );
+
+  assert.equal(loader.master.playlists[0].retries, 0, 'retries starts at zero on a sucessful load');
+
+  // playlist
+  this.requests.shift().respond(404);
+
+  assert.equal(loader.master.playlists[0].retries, 1, 'retries incremented by one on playlist load error');
+
+  loader.media(loader.master.playlists[1]);
+  loader.media(loader.master.playlists[0]);
+
+  this.requests[1].respond(
+    200, null,
+    '#EXTM3U\n' +
+    '#EXT-X-MEDIA-SEQUENCE:0\n' +
+    '#EXTINF:10,\n' +
+    '0.ts\n'
+  );
+  assert.equal(loader.master.playlists[0].retries, 0, 'retries resets to zero when a playlist sucessfully loads');
```

Comment:
Thanks for adding this case, an important one to check so that we don't increment forever.